filemanager/s3manager_v2.go

```diff
@@ -0,0 +1,324 @@
+package filemanager
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/http"
+	"net/url"
+	"os"
+	"path"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/smithy-go"
+
+	s3manager "github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/samber/lo"
+
+	"github.com/rudderlabs/rudder-go-kit/awsutil"
+	appConfig "github.com/rudderlabs/rudder-go-kit/config"
+	"github.com/rudderlabs/rudder-go-kit/logger"
+)
+
+const ServiceName = "s3"
+
+type S3ManagerV2 struct {
+	*baseManager
+	config *S3Config
+
+	sessionConfig *awsutil.SessionConfig
+	client        *s3.Client
+	clientMu      sync.Mutex
+}
+
+// NewS3ManagerV2 creates a new file manager for S3 using v2 aws sdk
+func NewS3ManagerV2(
+	config map[string]interface{}, log logger.Logger, defaultTimeout func() time.Duration,
+) (*S3ManagerV2, error) {
+	var s3Config S3Config
+	if err := mapstructure.Decode(config, &s3Config); err != nil {
+		return nil, err
+	}
+
+	sessionConfig, err := awsutil.NewSimpleSessionConfig(config, ServiceName)
+	if err != nil {
+		return nil, err
+	}
+
+	s3Config.RegionHint = appConfig.GetString("AWS_S3_REGION_HINT", "us-east-1")
```

Comment:
We do have [conf](https://github.com/rudderlabs/rudder-go-kit/blob/v0.15.12/filemanager/filemanager.go#L79) there, why are we accessing it the global way? Can't we use DI instead?